### PR TITLE
Don't use fluid layout value in typography calculation.

### PR DESCRIPTION
**Pull request by tellthemachines**
Trac ticket: https://core.trac.wordpress.org/ticket/58754

Adds the PHP changes from https://github.com/WordPress/gutenberg/pull/53551 and https://github.com/WordPress/gutenberg/pull/53554 to core.

To test: using Twenty Twenty Three or another block theme with fluid typography enabled, change the value of `settings.layout.wideSize` in theme.json to use a fluid value such as `clamp(1000px, 85vw, 2000px)`. Check that font sizes remain fluid in the front end. (For the editor part, an npm package update will be required; that will be done separately.)

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by ramonjd — APPROVED** [review]
Works for me. Thank you!

### Before

https://github.com/WordPress/wordpress-develop/assets/6458278/486f3e72-dca2-4429-a604-8239bcd595ba



### After

https://github.com/WordPress/wordpress-develop/assets/6458278/3ac40d4a-256e-4782-bfd3-0e4413c96c8d

Aside from the minor PHP lint warning (`Expected 1 space before "?"; 2 found`), I think the failing PHP test `Tests_Theme_ThemeDir::test_theme_list` is related. It's expecting `Block Theme Child Theme With Fluid Layout` in the array.

File: src/wp-includes/block-supports/typography.php [hunk 1]
```diff
@@ -517,7 +517,7 @@ function wp_get_typography_font_size_value( $preset, $should_use_fluid_typograph
 		: array();
 
 	// Defaults.
-	$default_maximum_viewport_width       = isset( $layout_settings['wideSize'] ) ? $layout_settings['wideSize'] : '1600px';
+	$default_maximum_viewport_width       = isset( $layout_settings['wideSize'] ) && ! empty( wp_get_typography_value_and_unit( $layout_settings['wideSize'] ) )  ? $layout_settings['wideSize'] : '1600px';
```
Comment (mukeshpanchal27) [inline src/wp-includes/block-supports/typography.php]:
```suggestion
	$default_maximum_viewport_width       = isset( $layout_settings['wideSize'] ) && ! empty( wp_get_typography_value_and_unit( $layout_settings['wideSize'] ) ) ? $layout_settings['wideSize'] : '1600px';
```

Fix PHPCS notice

**Comment by tellthemachines** [comment]
Thanks for the reviews folks! Looks like all the tests are passing now.

**Comment by tellthemachines** [comment]
Committed to trunk in[ r56503](https://core.trac.wordpress.org/changeset/56503); leaving this PR open for commit to the 6.3 release branch.

**Comment by tellthemachines** [comment]
Closing as this was committed to the 6.3 branch in https://core.trac.wordpress.org/changeset/56737

**Pull request closed without merging**